src/spoke/libraries/ReserveFlagsMap.sol

```diff
@@ -0,0 +1,110 @@
+  /// @param flags The current ReserveFlags.
+  /// @param status The new status for the `paused` flag.
+  /// @return The updated ReserveFlags.
+  function setPaused(ReserveFlags flags, bool status) internal pure returns (ReserveFlags) {
+    return ReserveFlags.wrap(_setFlag(ReserveFlags.unwrap(flags), PAUSED_MASK, status));
+  }
+
+  /// @notice Sets the new status for the `frozen` flag.
+  /// @param flags The current ReserveFlags.
+  /// @param status The new status for the `frozen` flag.
+  /// @return The updated ReserveFlags.
+  function setFrozen(ReserveFlags flags, bool status) internal pure returns (ReserveFlags) {
+    return ReserveFlags.wrap(_setFlag(ReserveFlags.unwrap(flags), FROZEN_MASK, status));
+  }
+
+  /// @notice Sets the new status for the `borrowable` flag.
+  /// @param flags The current ReserveFlags.
+  /// @param status The new status for the `borrowable` flag.
+  /// @return The updated ReserveFlags.
+  function setBorrowable(ReserveFlags flags, bool status) internal pure returns (ReserveFlags) {
+    return ReserveFlags.wrap(_setFlag(ReserveFlags.unwrap(flags), BORROWABLE_MASK, status));
+  }
+
+  /// @notice Sets the new status for the `receiveSharesEnabled` flag.
+  /// @param flags The current ReserveFlags.
+  /// @param status The new status for the `receiveSharesEnabled` flag.
+  /// @return The updated ReserveFlags.
+  function setReceiveSharesEnabled(
+    ReserveFlags flags,
+    bool status
+  ) internal pure returns (ReserveFlags) {
+    return
+      ReserveFlags.wrap(_setFlag(ReserveFlags.unwrap(flags), RECEIVE_SHARES_ENABLED_MASK, status));
+  }
+
+  /// @notice Returns the `paused` flag status.
+  /// @param flags The current ReserveFlags.
+  /// @return True if the flag is set.
+  function paused(ReserveFlags flags) internal pure returns (bool) {
+    return (ReserveFlags.unwrap(flags) & PAUSED_MASK) != 0;
+  }
+
+  /// @notice Returns the `frozen` flag status.
+  /// @param flags The current ReserveFlags.
+  /// @return True if the flag is set.
+  function frozen(ReserveFlags flags) internal pure returns (bool) {
+    return (ReserveFlags.unwrap(flags) & FROZEN_MASK) != 0;
+  }
+
+  /// @notice Returns the `borrowable` flag status.
+  /// @param flags The current ReserveFlags.
+  /// @return True if the flag is set.
+  function borrowable(ReserveFlags flags) internal pure returns (bool) {
+    return (ReserveFlags.unwrap(flags) & BORROWABLE_MASK) != 0;
+  }
+
+  /// @notice Returns the `receiveSharesEnabled` flag status.
+  /// @param flags The current ReserveFlags.
+  /// @return True if the flag is set.
+  function receiveSharesEnabled(ReserveFlags flags) internal pure returns (bool) {
+    return (ReserveFlags.unwrap(flags) & RECEIVE_SHARES_ENABLED_MASK) != 0;
+  }
+
+  /// @notice Sets the new status for the given flag.
+  function _setFlag(uint8 flags, uint8 mask, bool status) internal pure returns (uint8) {
+    return status ? flags | mask : flags & ~mask;
```

Comment:
we could also do
```
return getStatus(flags, mask) != status ? flags ^ mask : flags
```
probably the same from gas perspective?

Reply:
I'll try it